### PR TITLE
Add more standalone docs for cloud API

### PR DESCRIPTION
**Pull request by kbatuigas**
## Description

Includes more details on cloud error codes.
Resolves https://github.com/redpanda-data/documentation-private/issues/2455
Review deadline: **30 May 2024**

## Page previews

[API quickstart](https://deploy-preview-522--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/cloud/api/cloud-api-quickstart/) - includes "Try the API" section
[Error and status codes](https://deploy-preview-522--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/cloud/api/cloud-api-errors/)

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: local-antora-playbook.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ content:
   - url: .
     branches: HEAD
   - url: https://github.com/redpanda-data/documentation
-    branches: [v/*, api, shared, site-search]
+    branches: [v/*, 2461_cloud-api-reference-edits, shared, site-search]
```
Comment (kbatuigas) [inline local-antora-playbook.yml]: NOTE: This needs to be changed back to `api` before merging.

**Review by towfiqa — APPROVED** [review]
Looks great. Love it @kbatuigas! Thank you.

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+| REASON_REMOTE_REPLICA_INVALID_TYPE | Remote read replicas cluster has invalid type.
+ 
+| REASON_REMOTE_REPLICA_SAME_AS_SOURCE | Remote read replica cannot be the same as the source cluster.
+ 
+| REASON_REMOTE_REPLICA_CYCLE | Remote read replica may not list the source cluster as a replica of itself.
+ 
+| REASON_REMOTE_REPLICA_INVALID_DELETION | Source cluster cannot be deleted if it has read replicas. |
+|===
+
+=== Data Plane API error details
```
Comment (kbatuigas) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc:150]: @towfiqa is there someone on the API team who can provide these descriptions?

Comment (towfiqa) [reply modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc:150]: @birdayz or @sago2k8 can you help here?

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+|=== 
+| Reason | Description 
+
+| REASON_UNSPECIFIED | 
+
+| REASON_FEATURE_NOT_CONFIGURED |
```
Comment (bojand) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc]:
```suggestion
| REASON_FEATURE_NOT_CONFIGURED | The feature is not configured.
```

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+=== Data Plane API error details
+
+|=== 
+| Reason | Description 
+
+| REASON_UNSPECIFIED | 
+
+| REASON_FEATURE_NOT_CONFIGURED | 
+
+| REASON_CONSOLE_ERROR |
```
Comment (bojand) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc]:
```suggestion
| REASON_CONSOLE_ERROR | Internal Redpanda Console or data plane error.
```

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+|=== 
+| Reason | Description 
+
+| REASON_UNSPECIFIED | 
+
+| REASON_FEATURE_NOT_CONFIGURED | 
+
+| REASON_CONSOLE_ERROR | 
+
+| REASON_REDPANDA_ADMIN_API_ERROR |
```
Comment (bojand) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc]:
```suggestion
| REASON_REDPANDA_ADMIN_API_ERROR | Redpanda Admin API returned an error.
```

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+| REASON_UNSPECIFIED | 
+
+| REASON_FEATURE_NOT_CONFIGURED | 
+
+| REASON_CONSOLE_ERROR | 
+
+| REASON_REDPANDA_ADMIN_API_ERROR | 
+
+| REASON_KAFKA_API_ERROR |
```
Comment (bojand) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc]:
```suggestion
| REASON_KAFKA_API_ERROR | Redpanda or Kafka protocol error.
```

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+| REASON_FEATURE_NOT_CONFIGURED | 
+
+| REASON_CONSOLE_ERROR | 
+
+| REASON_REDPANDA_ADMIN_API_ERROR | 
+
+| REASON_KAFKA_API_ERROR | 
+
+| REASON_KAFKA_CONNECT_API_ERROR |
```
Comment (bojand) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc]:
```suggestion
| REASON_KAFKA_CONNECT_API_ERROR | Kafka Connect API error.
```

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+| REASON_CONSOLE_ERROR | 
+
+| REASON_REDPANDA_ADMIN_API_ERROR | 
+
+| REASON_KAFKA_API_ERROR | 
+
+| REASON_KAFKA_CONNECT_API_ERROR | 
+
+| REASON_TYPE_MAPPING_ERROR |
```
Comment (bojand) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc]:
```suggestion
| REASON_TYPE_MAPPING_ERROR | Type mapping error translating internal or external types to API types.
```

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+| REASON_REDPANDA_ADMIN_API_ERROR | 
+
+| REASON_KAFKA_API_ERROR | 
+
+| REASON_KAFKA_CONNECT_API_ERROR | 
+
+| REASON_TYPE_MAPPING_ERROR | 
+
+| REASON_SECRET_STORE_ERROR |
```
Comment (bojand) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc]:
```suggestion
| REASON_SECRET_STORE_ERROR | Cloud provider's secret store manager error.
```

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 10]
```diff
@@ -0,0 +1,142 @@
+= Cloud API Error and Status Codes
+:description: Error and status codes you might encounter when using the Cloud API.
+:page-cloud: true
+:page-api: cloud
+:page-beta: true
+
+The Redpanda Cloud API uses HTTP codes to indicate the status of a request. The response payload also includes <<error-codes-and-details,additional error codes and descriptions>> that provide more detail about why an operation failed.
+
+== HTTP response codes
```
Comment (birdayz) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc:41]: can we show an example here?

eg if i do
```
curl https://api.redpanda.com/clusters/v1beta2 | jq
```

i get a response 

```
{
  "code": "UNAUTHENTICATED",
  "message": "Bearer token is not present",
  "details": [
    {
      "@type": "google.rpc.ErrorInfo",
      "reason": "REASON_NO_TOKEN",
      "domain": "redpanda.com/controlplane",
      "metadata": {}
    },
    {
      "@type": "google.rpc.Help",
      "links": [
        {
          "description": "Please ensure you have a valid token to authenticate against the API. If you don't have a token, create a client and follow the instructions to request a token.",
          "url": "https://cloud.redpanda.com/clients"
        }
      ]
    }
  ]
}
```

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+ 
+| REASON_REMOTE_REPLICA_INVALID_DELETION | Source cluster cannot be deleted if it has read replicas. |
+|===
+
+=== Data Plane API error details
+
+|=== 
+| Reason | Description 
+
+| REASON_UNSPECIFIED |
```
Comment (birdayz) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc]: you can remove UNSPECIFIED. we will never return it.

File: modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+The response payload contains additional error details: a https://grpc.io/docs/guides/status-codes/[gRPC status code], and an error message that contains further details. 
+
+The Control Plane API returns a set of possible error codes that are mostly distinct from the possible Data Plane API error codes.
+
+=== Control Plane API error codes
+
+|=== 
+| Reason | Description 
+
+| REASON_UNSPECIFIED |
```
Comment (birdayz) [inline modules/deploy/pages/deployment-option/cloud/api/cloud-api-errors.adoc]: you can remove UNSPECIFIED. we will never return it.

**Review by birdayz — APPROVED** [review]

**Review by Deflaimun — APPROVED** [review]

**Review by Feediver1 — APPROVED** [review]
lgtm

**Pull request merged**